src/scripts/main.js

```diff
@@ -1,3 +1,293 @@
+      type="text"
+      required
+      data-qa="name"
+    >
+  </label>
+  <label>
+    Position:
+    <input
+      name="position"
+      type="text"
+      required
+      data-qa="position"
+    >
+  </label>
+  <label>
+    Office:
+    <select name="office" required data-qa="office">
+      <option>Tokyo</option>
+      <option>Singapore</option>
+      <option>London</option>
+      <option>New York</option>
+      <option>Edinburgh</option>
+      <option>San Francisco</option>
+    </select>
+  </label>
+  <label>
+    Age:
+    <input
+      name="age"
+      type="number"
+      required
+      data-qa="age"
+    >
+  </label>
+  <label>
+    Salary:
+    <input
+      name="salary"
+      type="number"
+      required
+      data-qa="salary"
+    >
+  </label>
+  <button type="submit">
+    Save to table
+  </button>
+`);
+
+form.addEventListener('submit', e => {
+  e.preventDefault();
+
+  const data = new FormData(form);
+  const tr = document.createElement('tr');
+  const formatter = new Intl.NumberFormat('en-US');
+  const salary = `$${formatter.format(data.get('salary'))}`;
+
+  tr.innerHTML = `
+    <td>${data.get('name')}</td>
+    <td>${data.get('position')}</td>
+    <td>${data.get('office')}</td>
+    <td>${data.get('age')}</td>
+    <td>${salary}</td>
+  `;
+
+  for (const letter of data.get('name')) {
+    if ('0123456789'.includes(letter)) {
+      pushNotification(150, 10, 'Error', 'Your data is incorrect', 'error');
+
+      return;
+    }
+  }
+
+  for (const letter of data.get('position')) {
+    if ('0123456789'.includes(letter)) {
+      pushNotification(150, 10, 'Error', 'Your data is incorrect', 'error');
+
+      return;
+    }
+  }
+
+  if (data.get('name').length < 4
+    || data.get('age') < 18
+    || data.get('age') > 90) {
+    pushNotification(150, 10, 'Error', 'Your data is incorrect', 'error');
+
+    return;
+  }
+
+  tableBody.append(tr);
+  arr.push(tr);
+
+  pushNotification(150, 10, 'Success',
+    'Your data added to the table.', 'success');
+});
+
+const pushNotification = (posTop, posRight, title, description, type) => {
+  const message = document.createElement('div');
+  const messageTitle = document.createElement('h2');
+  const paragraph = document.createElement('p');
+
+  message.className = `notification ${type}`;
+  message.style.boxSizing = 'content-box';
+  message.style.top = posTop + 'px';
+  message.style.right = posRight + 'px';
+  document.body.append(message);
+
+  messageTitle.className = 'title';
+  messageTitle.textContent = title;
+  message.append(messageTitle);
+
+  paragraph.className = 'description';
+  paragraph.textContent = description;
+  message.append(paragraph);
+
+  setTimeout(() => message.remove(), 2000);
+};
+
+// Change text in cell
+tableBody.addEventListener('dblclick', e => {
+  if (e.target.tagName !== 'TD') {
+    return;
+  }
+
+  previousText = e.target.textContent;
+
+  input.className = 'cell-input';
+
+  e.target.textContent = '';
```

Comment:
why you ignored this comment? if you have some question feel free to ask in fe_chat